sudo/src/main.rs

```diff
@@ -123,13 +124,27 @@ fn sudo_process() -> Result<std::convert::Infallible, Error> {
     let current_env = std::env::vars_os().collect();
     let target_env = environment::get_target_environment(current_env, &context, &policy);
 
+    let pid = context.process.pid;
+
     // run command and return corresponding exit code
-    Ok(sudo_exec::run_command(context, target_env)?)
+    let (reason, emulate_default_handler) = sudo_exec::run_command(context, target_env)?;
+
+    // Run any clean-up code before this line.
+    emulate_default_handler.store(true, Ordering::Relaxed);
```

Comment:
this should use `Ordering::Release` or even `Ordering::SeqCst` to synchronize with the signal handler that [reads the atomic using `Ordering::SeqCst`](https://docs.rs/signal-hook/0.3.15/src/signal_hook/flag.rs.html#224)